src/vehicle.h

```diff
@@ -221,6 +226,9 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
     item base;
     std::list<item> items; // inventory
 
+    /** read only from legacy savegames */
+    int legacy_hp = -1;
```

Comment:
Why do you need to keep `legacy_hp` (especially in the public section)?  Why don't damage `base` using the read value and forget about the legacy?